test/src/test/java/org/corfudb/integration/MergeTestListenerIT.java

```diff
@@ -0,0 +1,402 @@
+
+        // Open tables on both side
+        int numTables = 2;
+        // Source
+        openGroupATable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcTablesGroupA);
+        // Sink
+        openGroupATable(numTables, sinkCorfuStores.get(SINK2_INDEX), sinkTablesGroupA);
+        openLocalTable(numTables, sinkCorfuStores.get(SINK2_INDEX), sinkLocalTables);
+        openMergedTable(sinkCorfuStores.get(SINK2_INDEX));
+
+        writeDataOnSource(0, NUM_WRITES);
+
+        startReplicationServers();
+
+        verifySessionInLogEntrySyncState(SINK2_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), NUM_WRITES, sinkTablesGroupA);
+
+        writeDataOnSink(NUM_WRITES, NUM_WRITES);
+
+        MergeTestListener listenerSink2 = new MergeTestListener(sinkCorfuStores.get(SINK2_INDEX)
+                , NAMESPACE, numTables,
+                numTables);
+
+        // Subscribe the listeners on the sink side
+        sinkCorfuStores.get(SINK2_INDEX).subscribeLogReplicationListener(listenerSink2,
+                NAMESPACE, STREAMING_TAG, sinkCorfuStores.get(SINK2_INDEX));
+
+        // Verify the listeners get all the update
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 2 * NUM_WRITES,
+                        Collections.singletonList(mergedTable));
+
+        writeDataOnSource(2*NUM_WRITES, NUM_WRITES);
+
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 3 * NUM_WRITES, sinkTablesGroupA);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 3 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+
+        writeDataOnSink(3*NUM_WRITES, NUM_WRITES);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES, sinkTablesGroupA);
+
+        logicalGroupClient.addDestinations(GROUP_A,
+                Collections.singletonList(DefaultClusterConfig.getSinkClusterIds().get(SINK3_INDEX)));
+        List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> sinkTablesGroupAOnSink3 = new ArrayList<>();
+        openGroupATable(numTables, sinkCorfuStores.get(SINK3_INDEX), sinkTablesGroupAOnSink3);
+
+        writeDataOnSource(NUM_WRITES, NUM_WRITES);
+
+        verifyGroupATableData(sinkCorfuStores.get(SINK3_INDEX), 3 * NUM_WRITES, sinkTablesGroupAOnSink3);
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES, sinkTablesGroupA);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+
+        logicalGroupClient.removeDestinations(GROUP_A,
+                Collections.singletonList(DefaultClusterConfig.getSinkClusterIds().get(SINK3_INDEX)));
+
+        writeDataOnSource(3*NUM_WRITES, NUM_WRITES);
```

Comment:
Didn't understand why do these 2 writes on Source? especially they wrote same key to LR table as Sink side local table?

Reply:
For line 153, I do this write on Source for the completion of the verification. Since we have written entries 0-499, 1000-1499 to the `GROUP_A` tables on Source side before line 153, and we also haven't written any entry to sink3's local table, if we do the verification of ` GROUP_A` tables on Sink3, it will throw NPE(the 500-999 entries is missing) If we do this write on Source, it won't throw NPE when `verifyGroupATableData` reads data 500-999.

For line 163, I think it's okay to remove it.